rust-keybroker/keybroker-app/src/main.rs

```diff
@@ -1,6 +1,58 @@
 // Copyright 2024 Contributors to the Veraison project.
 // SPDX-License-Identifier: Apache-2.0
 
+use clap::Parser;
+use keybroker_client::error::Error as KeybrokerError;
+use keybroker_client::{CcaExampleToken, KeyBrokerClient, TsmAttestationReport};
+
+/// Structure for parsing and storing the command-line arguments
+#[derive(Clone, Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    /// The base url at which the keybroker server listens
+    #[arg(short, long, default_value = "http://127.0.0.1")]
+    base_url: String,
+
+    /// The port on which the keybroker server listens
+    #[arg(short, long, default_value_t = 8088)]
+    port: u16,
+
+    /// The key to use
+    #[arg(short, long, default_value = "skywalker")]
+    key_name: String,
+
+    /// Use a CCA example token (instead of the TSM report)
+    #[arg(short, long, default_value_t = false)]
+    mock_evidence: bool,
+
+    /// Set the application verbosity
+    #[arg(short, long, default_value_t = false)]
+    verbose: bool,
+}
+
 fn main() {
-    println!("Hello, world!");
+    let args = Args::parse();
+
+    let client = KeyBrokerClient::new(&args.base_url, args.port, args.verbose);
+
+    let attestation_result = if args.mock_evidence {
+        client.get_key(&args.key_name, &CcaExampleToken {})
+    } else {
+        client.get_key(&args.key_name, &TsmAttestationReport {})
+    };
+
+    match attestation_result {
+        Ok(key) => {
+            let plainstring_key = String::from_utf8(key).unwrap();
+            println!("Attestation success :-) ! The key returned from the keybroker is '{plainstring_key}'")
+        }
+
+        Err(error) => {
+            if let KeybrokerError::AttestationFailure(reason, details) = error {
+                println!("Attestation failure :-( ! {reason}: {details}")
+            } else {
+                panic!("The key request failed with: {error:?}");
```

Comment:
Ideally, I would have preferred the program to behave like a standard unix program: return 0 on attestation success, 1 on attestation failure and some other code for other errors, so it can be used in shell scripts for example / can be easily/programmatically tested. The advantage I saw in `panic` is that you can not miss its side effects, my lack of idiomatic Rust blocked me there.

Reply:
It looks like [`std::process::exit`](https://doc.rust-lang.org/nightly/std/process/fn.exit.html) may do the job.

Reply:
Resolving this conversation as `keybroker-app` now returns with an exit code.